build-scripts/build_compiler.js

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2018 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const {spawn, spawnSync} = require('child_process');
+const ncp = require('ncp');
+const Semver = require('semver');
+const version = require('../lerna.json').version;
+const fs = require('fs');
+const packageVer = new Semver(version);
+const glob = require('glob');
+
+const gitCmd = spawnSync('git', ['tag', '--points-at', 'HEAD'], {
+  cwd: './compiler'
+});
+let compilerVersion = '1.0-SNAPSHOT';
+if (gitCmd.status === 0) {
+  const currentTag = gitCmd.stdout.toString().replace(/\s/g, '');
+  let normalizedTag = currentTag;
+  if (normalizedTag) {
+    normalizedTag = currentTag.replace(/^([a-z]+-)?v\d{8}(.*)$/,
+        (match, g1, g2) => match.substr((g1 || '').length, match.length - (g1 || '').length - (g2 || '').length));
+  }
+
+  // If the compiler submodule is pointing at a tagged version that matches the package
+  // major version, update the compiler build to use the correct version number.
+  if (normalizedTag === `v${packageVer.major}`) {
+    compilerVersion = normalizedTag;
+    const pomPaths = glob.sync('./compiler/**/pom*.xml');
+    pomPaths.forEach(pomPath => {
+      const contents = fs.readFileSync(pomPath, 'utf8');
+      fs.writeFileSync(pomPath, contents.replace('<version>1.0-SNAPSHOT</version>', `<version>${compilerVersion}</version>`), 'utf8');
+    });
```

Comment:
It strikes me as fishy to modify the compiler git submodule like this.

We currently create a a maven-vYYYYMMDD branch based on the vYYYYMMDD tag and edit the pom files in there.
I think this script should maybe be checking out that branch instead of hacking the files itself.

Reply:
Well that's an interesting concept. I'll give that a try. I really wish the release tag just pointed at these branches instead - seems pointless to have both.

Reply:
Hmm - I just realized that's going to screw up some tests: https://github.com/google/closure-compiler-npm/blob/mono-repo/test/compiler.js#L78-L96

I agree this is not a great process. Let's talk about options at the next meeting.

Reply:
We can definitely discuss this at the next group meeting.
In the meantime...

The point of having both is that the tag records exactly where we cut the release, while the maven branch keeps the external release changes we don't want synced back to Google's internal repo segregated.

If you're comfortable modifying the local repo, why not just check out the corresponding maven branch?
If there isn't one, then back off and leave `SNAPSHOT-1.0` in there.

Reply:
That's some of the info I was missing. The maven release process is totally opaque to me right now - so I don't know any of the "why". In other projects, I'm used to seeing the tagged release basically be on what the compiler has as the maven release branch.

I can try modifying and checking out the branch. I'll have to rework the tests a bit though.

Reply:
I gave this a try, and ran into an issue:

The maven release branch isn't currently changing the pom-gwt.xml file, so the gwt version doesn't build here. I'll have to modify pom.xml files anyway.

https://github.com/google/closure-compiler/compare/v20181008...closure-compiler-parent-v20181008

Reply:
This works now!